scripts/servers.py

```diff
@@ -369,31 +369,35 @@ def managed_redis_server() -> Iterator[psutil.Process]:
 
 def create_managed_web_browser(
     port: int
-) -> Optional[ContextManager[psutil.Process]]:
-    """Returns a context manager for a web browser targeting the given port on
+) -> ContextManager[psutil.Process]:
+    """Returns a ContextManager for a web browser targeting the given port on
     localhost. If a web browser cannot be opened on the current system by Oppia,
-    then returns None instead.
+    then raises an exception.
 
     Args:
         port: int. The port number to open in the web browser.
 
     Returns:
-        context manager|None. The context manager to a web browser window, or
-        None if the current operating system does not support web browsers.
+        ContextManager. The ContextManager to a web browser window if the
+        current operating system can be identified and a web browser can be
+        launched automatically.
+
+    Raises:
+        Exception. Unable to launch the web browser (this happens when the
+            Operating System cannot be identified).
     """
     url = 'http://localhost:%s/' % port
     human_readable_name = 'Web Browser'
     if common.is_linux_os():
-        if any(re.match('.*VBOX.*', d) for d in os.listdir('/dev/disk/by-id/')):
-            return None
-        else:
-            return managed_process(
-                ['xdg-open', url], human_readable_name=human_readable_name)
+        return managed_process(
+            ['xdg-open', url], human_readable_name=human_readable_name)
```

Comment:
Sorry, I forgot to mention this. Can you add a test for this too?

Reply:
Isn't the below test for this?

`    def test_managed_web_browser_on_linux_os(self) -> None:
        popen_calls = self.exit_stack.enter_context(self.swap_popen())
        self.exit_stack.enter_context(self.swap(common, 'OS_NAME', 'Linux'))
        self.exit_stack.enter_context(self.swap_to_always_return(
            os, 'listdir', value=[]))

        managed_web_browser = servers.create_managed_web_browser(123)
        self.assertIsNotNone(managed_web_browser)
        assert managed_web_browser is not None
        self.exit_stack.enter_context(managed_web_browser)

        self.assertEqual(len(popen_calls), 1)
        self.assertEqual(
            popen_calls[0].program_args, ['xdg-open', 'http://localhost:123/'])`